mantidproject/mantid
# Create Stitching tab in Options Dialog for Reflectometry Polref interface

The user can now specify their own Start/End overlaps and Params through the stitching tab as well as choosing to scale the LHS workspace when using Stitch1DMany.

**To test:**
There is a `.tbl` file containing SURF runs [here](https://slack-files.com/T02J4MMEW-F0ST0CPQA-6cf6dbdaf1) which should be used for testing.

**_No Changes**_
- Open the reflectometry interface (`Interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Load the SURF nxs file into Mantid
- Load the SURF file into the interface (`Reflectometry > Open Table`)
- Go to the stitching tab (`Reflectometry > Options > Stitching`)
- Ensure that all checkboxes are disabled by default and leave them.
- Click Process and ensure that the stitched workspace has extents that are similar to the QMin and QMax found in the table.

**_Change StartOverlaps**_
- Open the reflectometry interface (`Interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Load the SURF .tbl file into the interface (`Reflectometry > Import Table`)
- Go to the stitching tab (`Reflectometry > Options > Stitching`)
- Enable the `StartOverlaps` checkbox and ensure that the LineEdit beneath it is enabled now.
- Enter some 2 startOverlaps, you can base these from the QMin found in the table from the previous case
- Ensure that the stitched workspace has the minimum extent value of the first start overlap you provided.

**_Change EndOverlaps**_
- Open the reflectometry interface (`Interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Load the SURF .tbl file into the interface (`Reflectometry > Import Table`)
- Go to the stitching tab (`Reflectometry > Options > Stitching`)
- Disable the `StartOverlaps` checkbox and ensure that the LineEdit beneath it is disabled now.
- Enable the `EndOverlaps` checkbox and ensure that the LineEdit beneath it is enabled now.
- Enter some 2 EndOverlaps, you can base these from the QMax found in the table from the previous case
- Ensure that the stitched workspace has the maximum extent value of the last end overlap you provided.

**_Change Params**_
- Open the reflectometry interface (`Interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Load the SURF .tbl file into the interface (`Reflectometry > Import Table`)
- Go to the stitching tab (`Reflectometry > Options > Stitching`)
- Disable the `EndOverlaps` checkbox and ensure that the LineEdit beneath it is disabled now.
- Enable the `Params` checkbox and ensure that the LineEdit beneath it is enabled now.
- Enter some 3 comma separated values e.g "1.0,0.5,2.0", you can base these from the QMin, QMax and DQ/q found in the table from the previous case.
- Ensure that the stitched workspace has the extents value of the first and last values that you entered for your Params.

Fixes #14848 .



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
